### ISSUE TITLE
Documentation experiences of an absolute Flux-beginner 

### ISSUE THREAD

**Issue by baggepinnen** [FluxML/Flux.jl#45]
I have been looking forward to playing with Flux.jl for a while and decided now is the time. I summarize my first experiences from reading through the docs, to hopefully highlight some potential improvements to what I realize is a very early version of documentation.
- The documentation introduction sometimes writes affine layers as `x*W` and sometimes as `W*x` which confuses the reader (at least me) about input dimensions.
- I figured out that `mymodel[1:3](x)` would calculate the output of the third layer. This is really convenient, but I couldn't find it in the docs. I am also interested in knowing how I calculate all (or some) intermediate layer activations without calculating anything twice. As in Tensorflow `run(session, [l1, l2, output], ...)`
- I can't find any info about how to specify number of training epochs and optimizer etc. I don't even know which optimizer is being used. (`Flux.train!` tells me how to specify number of epochs, but I don't know if there's a default value or it is determiend automatically based on progress)
- In the MNIST example, I can't determine what loss function is minimized. Is there an automatic choice being made? Maybe based on the last layer in the model? 

I find Flux very interesting, especially after having watched a presentation of Flux somewhere on YouTube, and hope to be able to contribute some time in the future!

**Comment by MikeInnes** [comment]
Thanks a lot for writing this up. I've done a big refactor of both Flux and its docs, so I think all of these are much clearer.

To address your second point, in general you'd have to to this however you'd normally do it in a Julia program, perhaps by returning a tuple. In the `Chain` case you can do:

```julia
l1 = mymodel[1](x)
output = mymodel[2:end](x)
```

**Comment by MikeInnes** [comment]
Closing this for now, but do let me know of any other feedback – thanks again!
